manifests/dev/vsphere-7.0u1/vanilla/rbac/vsphere-csi-controller-rbac.yaml

```diff
@@ -35,7 +35,7 @@ rules:
     verbs: ["create", "get", "list", "watch", "update", "delete"]
   - apiGroups: ["apiextensions.k8s.io"]
     resources: ["customresourcedefinitions"]
-    verbs: ["get", "create"]
+    verbs: ["get", "create", "update"]
```

Comment:
We do not need `update` for release 2.1 code.

Reply:
But what if we want to run latest CSI images on 7.0U1 vsphere release ?
We still need it, right ?

Reply:
Yes, future releases will require if we change CRD spec.

Can we enhance https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/kubernetes/kubernetes.go#L337-L353

We can compare current spec with the new spec, and onlly issue update when we observe spec has changed and needs to be updated. 

This way we can reduce the additional `update` rbac rule.

Reply:
@divyenpatel 
I tried comparing the specs and observed that the old and new spec are different when update was invoked:
```
{Group:cns.vmware.com Version:v1alpha1 Names:{Plural:cnsvspherevolumemigrations Singular:cnsvspherevolumemigration ShortNames:[] Kind:CnsVSphereVolumeMigration ListKind:CnsVSphereVolumeMigrationList Categories:[]} Scope:Cluster Validation:nil Subresources:nil Versions:[{Name:v1alpha1 Served:true Storage:true Schema:nil Subresources:nil AdditionalPrinterColumns:[]}] AdditionalPrinterColumns:[] Conversion:&CustomResourceConversion{Strategy:None,WebhookClientConfig:nil,ConversionReviewVersions:[],} PreserveUnknownFields:0xc000450ef4}
```

```
{Group:cns.vmware.com Version: Names:{Plural:cnsvspherevolumemigrations Singular:cnsvspherevolumemigration ShortNames:[] Kind:CnsVSphereVolumeMigration ListKind: Categories:[]} Scope:Cluster Validation:nil Subresources:nil Versions:[{Name:v1alpha1 Served:true Storage:true Schema:nil Subresources:nil AdditionalPrinterColumns:[]}] AdditionalPrinterColumns:[] Conversion:nil PreserveUnknownFields:<nil>}
```

I don't think comparing will help here to decide whether or not to invoke update

Reply:
@chethanv28 Can you try to compare the diff? We have not changed the CRD spec, so why we need to update it unnecessarily unless we are upgrading the API version or something in the spec.

cc: @BaluDontu

Reply:
Let's not compare whether to update or not. That's a little more complex. But we have update code for CRD. If it goes to 2.1 release branch then we need update permission.

Reply:
@divyenpatel Per our discussion, we will still need the update rule for CRD in vanilla rbac file because of the way CRD is created and updated: https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/kubernetes/kubernetes.go#L340